src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -265,19 +265,10 @@ public void WaitForCancelCompletion()
                 {
                     taskCleanedUp = _requestTask.Wait(BuildParameters.RequestBuilderShutdownTimeout);
                 }
-                catch (AggregateException e)
+                catch (AggregateException e) when (e.Flatten().InnerExceptions.All(ex => (ex is TaskCanceledException || ex is OperationCanceledException)))
```

Comment:
Nit: easier to read this way in my opinion. I haven't seen this c# feature before, so I'd like to say that this suggested syntax is the way we do it 🙂 maybe with the `when` being indented a tab?
```suggestion
                catch (AggregateException e)
                when (e.Flatten().InnerExceptions.All(ex => (ex is TaskCanceledException || ex is OperationCanceledException)))
```

Either way, nice cleanup here.

Reply:
We actually have a lot of examples with when on the same line as the catch. See BuildManager line 1134, 1139, 1165, 1353, and 1362 or BuildRequestEngine line 283, for example. Also, I like it slightly better on one line.